BuildPropertiesPlugin/plugin/src/main/groovy/com/novoda/buildproperties/BuildPropertiesPlugin.groovy

```diff
@@ -0,0 +1,170 @@
+package com.novoda.buildproperties
+
+import org.gradle.api.GradleException
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+
+class BuildPropertiesPlugin implements Plugin<Project> {
+
+    private final Map<String, BuildProperties> propertiesMap = [:]
+
+    @Override
+    void apply(Project project) {
+        boolean isAndroidApp = project.plugins.hasPlugin('com.android.application')
+        if (!isAndroidApp) {
+            throw new GradleException("This plugin can be applied only to an Android application project")
+        }
+
+        project.android.extensions.add('buildProperties', project.container(BuildProperties))
+        def container = project.android.extensions.getByName('buildProperties')
+        project.afterEvaluate {
+            container.all { BuildProperties buildProperties ->
+                Properties properties = new Properties()
+                properties.load(new FileInputStream(buildProperties.file))
```

Comment:
I would handle the loading logic inside the `BuildProperties` class, but can be done later.